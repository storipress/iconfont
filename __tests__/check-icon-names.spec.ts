import fs from 'node:fs/promises'
import path from 'node:path'
import { expect, it } from 'vitest'
import { parse } from 'postcss'

const ICON_PREFIX = '.icon-'

it('test icon names not change', async () => {
  const css = await fs.readFile(path.resolve(__dirname, '../iconfont.css'), 'utf8')
  const root = parse(css, { from: 'style.css' })
  const classNames: string[] = []
  root.walkRules((rule) => {
    for (const selector of rule.selectors) {
      if (selector.startsWith(ICON_PREFIX))
        classNames.push(selector.slice(ICON_PREFIX.length).replace(/(?:\s\.path\d)?:.*/, ''))
    }
  })
  const cleanedClassNames = [...new Set(classNames.sort((a, b) => a.localeCompare(b)))]
  expect(cleanedClassNames).toMatchSnapshot()
})
