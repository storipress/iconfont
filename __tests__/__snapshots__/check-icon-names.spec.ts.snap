// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test icon names not change 1`] = `
[
  "account_circle",
  "add_image",
  "adsense",
  "all_caps",
  "api",
  "arrow_left",
  "arrow_preview",
  "arrow_right",
  "arrow-down",
  "arrow-up",
  "article",
  "attachment",
  "billing",
  "bookmark",
  "boolean",
  "calendar",
  "calendar-done",
  "check",
  "chevron_down",
  "chevron_left",
  "chevron_right",
  "chevron_up",
  "circle-cancel",
  "circle-tick",
  "clipboard-tick",
  "clock",
  "code",
  "code_2",
  "code_injection",
  "Code-Injection",
  "codepen",
  "color",
  "comment",
  "copy",
  "cross",
  "cross_thin",
  "cross_thin-2",
  "database",
  "delete",
  "desk",
  "desktop",
  "disqus",
  "divider",
  "dots_horizontal",
  "dots_vertical",
  "down_arrow",
  "draft",
  "drag-handle",
  "edit_meta",
  "email",
  "facebook",
  "facebook-f",
  "file",
  "filter",
  "flag",
  "focal_point",
  "focus",
  "focus-2",
  "folder",
  "font",
  "for_review",
  "format_align_center",
  "format_align_left",
  "format_align_right",
  "format_align_stretch",
  "format_bold",
  "format_italics",
  "format_line_spacing",
  "format_underline",
  "fountain_pen_tip",
  "gallery",
  "github",
  "google",
  "google_analytics",
  "goto-url",
  "graphql",
  "H1",
  "H2",
  "H3",
  "hamburger",
  "history",
  "home",
  "html_embed",
  "image",
  "infinity",
  "info",
  "instagram",
  "integrations",
  "internet",
  "justify_expand",
  "justify_fill",
  "justify_fit",
  "lego",
  "light-bulb",
  "lightning",
  "link_break",
  "link_variant",
  "linkedin",
  "list_bullet",
  "list_number",
  "lock",
  "lock-filled",
  "logomark_L",
  "logomark_s",
  "logout",
  "lower_caps",
  "magic",
  "magic-2",
  "magnify",
  "mailchimp",
  "medium",
  "minus_circle",
  "minus_circle_outline",
  "mobile",
  "note",
  "number",
  "page_colour",
  "page_elements",
  "pen",
  "pinterest",
  "plus",
  "plus_circle",
  "plus_circle_outline",
  "preview",
  "publication",
  "publications",
  "published",
  "question-mark",
  "question-mark-inverse",
  "quote",
  "reddit",
  "redo",
  "refresh",
  "return",
  "rocket",
  "sad",
  "saving",
  "schedule",
  "selector",
  "SEO",
  "settings",
  "settings-2",
  "share",
  "shopify",
  "slack",
  "sort",
  "soundcloud",
  "speech",
  "spotify",
  "sprout-grow",
  "star",
  "star_outline",
  "steps",
  "strikethrough",
  "svg",
  "swap-replace",
  "table_of_contents",
  "table-of-contents",
  "tablet",
  "tag",
  "team",
  "templates",
  "termly",
  "text",
  "text_2",
  "text_colour",
  "text_hover_colour",
  "theme",
  "theme-2",
  "tick",
  "tick_circle",
  "tick-badge",
  "twitter",
  "undo",
  "unsplash",
  "up_arrow",
  "upload",
  "user",
  "users",
  "vimeo",
  "warning",
  "web",
  "webflow",
  "whatsapp",
  "youtube",
  "zapier",
]
`;
